Analysis/interface/GenLepton.h

```diff
@@ -74,18 +75,20 @@ class GenLepton {
     static constexpr size_t MaxNumberOfParticles = 1000;
 
     template<typename GenParticleT>
-    static std::vector<GenLepton> fromGenParticleCollection(const std::vector<GenParticleT>& gen_particles)
+    static std::vector<GenLepton> fromGenParticleCollection(const std::vector<GenParticleT>& gen_particles, bool isGeantPlusSim_)
     {
         std::vector<GenLepton> leptons;
         std::map<const GenParticleT*, int> processed_particles;
         for(const auto& particle : gen_particles) {
             if(processed_particles.count(&particle)) continue;
-            if(!(particle.statusFlags().isPrompt() && particle.statusFlags().isFirstCopy())) continue;
             const int abs_pdg = std::abs(particle.pdgId());
+            if(!((particle.statusFlags().isPrompt() && particle.statusFlags().isFirstCopy())
+                || (isGeantPlusSim_ && particle.status()==8 && abs_pdg==15))) continue;
```

Comment:
better to have `std::optional<int> geantParticleStatus = std::optional<int>()`;
and have comparison
```c++
const int abs_pdg = std::abs(particle.pdgId());
const GenParticle::PdgId pdg_id = static_cast<GenParticle::PdgId>(abs_pdg);
const bool isFirstPrompt = particle.statusFlags().isPrompt() && particle.statusFlags().isFirstCopy();
const bool isGeantTau = geantParticleStatus && particle.status() == *geantParticleStatus && pdg_id == GenParticle::PdgId::tau;
if(!(isFirstPrompt || isGeantTau)) continue;
if(!GenParticle::ChargedLeptons().count(pdg_id)) continue;
```